struct Test {
    init { self } // ok
    method1(self) => self // ok
    method2() { return self } // error 
    static1() => self // error
    static2() => self // error
}

self // error
do { self } // error
fn func() => self // error

%output

error: Cannot use `self` outside of an instance method or `init` block
  ┌─ <source: #4b446b236ae9cab2>:4:24
  │
4 │     method2() { return self } // error 
  │                        ^^^^

error: Cannot use `self` outside of an instance method or `init` block
  ┌─ <source: #4b446b236ae9cab2>:5:18
  │
5 │     static1() => self // error
  │                  ^^^^

error: Cannot use `self` outside of an instance method or `init` block
  ┌─ <source: #4b446b236ae9cab2>:6:18
  │
6 │     static2() => self // error
  │                  ^^^^

error: Cannot use `self` outside of an instance method or `init` block
  ┌─ <source: #4b446b236ae9cab2>:9:1
  │
9 │ self // error
  │ ^^^^

error: Cannot use `self` outside of an instance method or `init` block
   ┌─ <source: #4b446b236ae9cab2>:10:6
   │
10 │ do { self } // error
   │      ^^^^

error: Cannot use `self` outside of an instance method or `init` block
   ┌─ <source: #4b446b236ae9cab2>:11:14
   │
11 │ fn func() => self // error
   │              ^^^^